examples/component/http-password-checker/go.mod

```diff
@@ -0,0 +1,31 @@
+module http-password-checker
+
+go 1.23.0
+
+require (
+	github.com/stretchr/testify v1.10.0
+	github.com/wagslane/go-password-validator v0.3.0
+	go.bytecodealliance.org v0.5.0
+	go.bytecodealliance.org/cm v0.1.0
```

Comment:
I would prefer it if we stuck to [the version `go.wasmcloud.dev/component` specifies](https://github.com/wasmCloud/go/blob/main/component/go.mod#L7) and update all of the examples once we update the component